### PR TITLE
migrate contributor_id to contributor_account_address in smart contracts

### PR DESCRIPTION
**Pull request by AnthonyBuisset**
- 🗃️ migrate past contribution count from contributor id to contributor account
- ♻️ use input contributor_account_address to assign and compute past contribution count

### CODE REVIEW

File: contracts/onlydust/marketplace/core/contributions/library.cairo [hunk 1]
```diff
@@ -271,19 +271,18 @@ namespace contributions {
     // Assign a contributor to a contribution
     func assign_contributor_to_contribution{
         syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr
-    }(contribution_id: ContributionId, contributor_id: Uint256) {
+    }(contribution_id: ContributionId, contributor_account_address: felt) {
         let contribution_exists = contribution_access.exists(contribution_id);
         if (contribution_exists == 0) {
             let contract_address = contribution_id.inner;
-            let contributor_account = contributor_id.low;
-            contribution_contributor_.write(contribution_id, contributor_id);
```
Comment (Bernardstanislas) [inline contracts/onlydust/marketplace/core/contributions/library.cairo:279]: I don't understand why this is removed?

File: contracts/onlydust/marketplace/core/contributions/library.cairo [hunk 2]
```diff
@@ -295,14 +294,11 @@ namespace contributions {
...
         syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr
-    }(contribution_id: ContributionId) {
+    }(contribution_id: ContributionId, contributor_account_address: felt) {
         let contribution_exists = contribution_access.exists(contribution_id);
         if (contribution_exists == 0) {
             let contract_address = contribution_id.inner;
-            let (contributor_id) = contribution_contributor_.read(contribution_id);
-            let contributor_account = contributor_id.low;
-            IContribution.unassign(contract_address, contributor_account);
-            contribution_contributor_.write(contribution_id, Uint256(0, 0));
```
Comment (Bernardstanislas) [inline contracts/onlydust/marketplace/core/contributions/library.cairo:305]: Same here

File: contracts/onlydust/marketplace/core/contributions/library.cairo (hunk 1 above)
Comment (AnthonyBuisset) [reply contracts/onlydust/marketplace/core/contributions/library.cairo:279]: this is not needed as passed as parameter from now on (contributor account might be different than contributor_id during the migration)

File: contracts/onlydust/marketplace/core/contributions/library.cairo (hunk 2 above)
Comment (AnthonyBuisset) [reply contracts/onlydust/marketplace/core/contributions/library.cairo:305]: same reason, it is now passed as parameter

**Review by Bernardstanislas — APPROVED** [review]

**Pull request merged**